sizefashion/src/pages/SingleProductPage/SingleProductPage.tsx

```diff
@@ -0,0 +1,49 @@
+import { useEffect } from 'react'
+import { useParams } from 'react-router-dom'
+import { useProductsContext } from '../../context/products_context'
+import { ProductImages, Loading } from '../../components'
+import { BackToProductsButton } from './BackToProductsButton'
+import { SingleProductContent } from './SingleProductContent'
+import ErrorPage from '../ErrorPage'
+
+const SingleProductPage = () => {
+  const { slug } = useParams<{ slug: string }>()
+  const {
+    singleProduct,
+    fetchSingleProduct,
+    singleProductLoading,
+    singleProductError,
+    allProducts,
+  } = useProductsContext()
+
+  const { image } = { ...singleProduct }
+
+  // caso aconteça um refresh na tela de produto
+  useEffect(() => {
+    if (slug) {
+      fetchSingleProduct(slug)
+    }
+    // eslint-disable-next-line
+  }, [slug, allProducts])
```

Comment:
@otaaviostein if the intention is to rehydrate your component for when the user opens the single product page, this effect can safely become a custom hook.

eg.

```js
// Your current effect handler
useEffect(() => {
    if (slug) {
      fetchSingleProduct(slug)
    }
    // eslint-disable-next-line
}, [slug, allProducts])

// use-product-on-mount.ts
function useProductOnMount() {
  const { slug } = useParams<{ slug: string }>()
  const { fetchSingleProduct } = useProductsContext()

  useEffect(() => {
    if (slug) {
      fetchSingleProduct(slug)
    }
   }, [slug])
}

// your-page.ts
import useProductOnMount from './use-product-on-mount'

useProductOnMount()
```